components/tidb_query/src/codec/batch/lazy_column.rs

```diff
@@ -174,7 +174,7 @@ impl LazyBatchColumn {
                         vec.push(None);
                     }
                     for row_index in logical_rows {
-                        vec[*row_index] = raw_vec[*row_index].decode(field_type, time_zone)?;
+                        vec[*row_index] = raw_vec[*row_index].decode(field_type, ctx)?;
```

Comment:
It's better to change the `decode` signature and put the `ctx` as first argument.

Reply:
I just wrap the original `Tz` with `ctx`. if we do need to unify the code, we could open another pr for it.